lib/trino-parquet/src/main/java/io/trino/parquet/writer/valuewriter/TimestampMillisValueWriter.java

```diff
@@ -14,31 +14,52 @@
 package io.trino.parquet.writer.valuewriter;
 
 import io.trino.spi.block.Block;
+import io.trino.spi.type.TimestampType;
 import io.trino.spi.type.Type;
 import org.apache.parquet.column.values.ValuesWriter;
 import org.apache.parquet.schema.PrimitiveType;
+import org.joda.time.DateTimeZone;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static io.trino.spi.type.Timestamps.MICROSECONDS_PER_MILLISECOND;
 import static java.lang.Math.floorDiv;
 import static java.util.Objects.requireNonNull;
+import static org.joda.time.DateTimeZone.UTC;
 
 public class TimestampMillisValueWriter
         extends PrimitiveValueWriter
 {
-    private final Type type;
+    private final TimestampType timestampType;
+    private final DateTimeZone parquetTimeZone;
 
-    public TimestampMillisValueWriter(ValuesWriter valuesWriter, Type type, PrimitiveType parquetType)
+    public TimestampMillisValueWriter(ValuesWriter valuesWriter, Type type, PrimitiveType parquetType, DateTimeZone parquetTimeZone)
     {
         super(parquetType, valuesWriter);
-        this.type = requireNonNull(type, "type is null");
+        requireNonNull(type, "type is null");
+        checkArgument(
+                type instanceof TimestampType && ((TimestampType) type).getPrecision() <= 9,
+                "type %s is not a TimestampType with precision <= 9",
+                type);
+        this.timestampType = (TimestampType) type;
```

Comment:
Why is this change needed ?

Reply:
Had adapted it from INT96 writer. Removed.